src/marqo/core/unstructured_vespa_index/unstructured_document.py

```diff
@@ -0,0 +1,180 @@
+            for chunk, embedding in zip(self.vespa_chunks, embeddings_list):
+                if "::" not in chunk:
+                    raise ValueError(f"Chunk {chunk} does not have a field_name::content format")
+                field_name, content = chunk.split("::", 2)
+                if field_name not in marqo_document[index_constants.MARQO_DOC_TENSORS]:
+                    marqo_document[index_constants.MARQO_DOC_TENSORS][field_name] = dict()
+                    marqo_document[index_constants.MARQO_DOC_TENSORS][field_name][index_constants.MARQO_DOC_CHUNKS] = []
+                    marqo_document[index_constants.MARQO_DOC_TENSORS][field_name][
+                        index_constants.MARQO_DOC_EMBEDDINGS] = []
+                marqo_document[index_constants.MARQO_DOC_TENSORS][field_name][index_constants.MARQO_DOC_CHUNKS].append(
+                    content)
+                marqo_document[index_constants.MARQO_DOC_TENSORS][field_name][
+                    index_constants.MARQO_DOC_EMBEDDINGS].append(embedding)
+
+        if self.vespa_multimodal_params:
+            marqo_document[unstructured_common.MARQO_DOC_MULTIMODAL_PARAMS] = dict()
+            for multimodal_field_name, serialized_multimodal_params in self.vespa_multimodal_params.items():
+                marqo_document[unstructured_common.MARQO_DOC_MULTIMODAL_PARAMS][multimodal_field_name] = \
+                    json.loads(serialized_multimodal_params)
+
+        if return_highlights and self.match_features:
+            marqo_document[index_constants.MARQO_DOC_HIGHLIGHTS] = self._extract_highlights()
+
+        return marqo_document
+
+    def _extract_highlights(self):
+        if not self.match_features:
+            raise ValueError("No match features found in the document")
+        try:
+            chunk_index: int = int(list(self.match_features[f"closest({unstructured_common.VESPA_DOC_EMBEDDINGS})"] \
+                                            ["cells"].keys())[0])
+        except KeyError:
+            raise ValueError("No match features found in the document")
+        field_name, content = self.vespa_chunks[chunk_index].split("::", 2)
+        return {field_name: content}
+
+
+class UnstructuredIndexDocument(BaseModel):
+    """A helper class to handle the conversion between Vespa and Marqo documents for an unstructured index.
+
+    The object can be instantiated from a Marqo document using the from_marqo_document method,
+    or can be instantiated from a Vespa document using the from_vespa_document method.
+    """
+    id: str = Field(alias=index_constants.VESPA_DOC_ID)
+    fields: UnstructuredVespaDocumentFields = Field(default_factory=UnstructuredVespaDocumentFields,
+                                                    alias=index_constants.VESPA_DOC_FIELDS)
+
+    class Config:
+        allow_population_by_field_name = True
+
+    @classmethod
+    def from_vespa_document(cls, document: Dict) -> "UnstructuredIndexDocument":
```

Comment:
Why do you need this? Shouldn't `UnstructuredIndexDocument(**vespa_doc_dict)` work? i.e. native pydantic deserialization